### ISSUE TITLE
FanControl blocks Windows shutdown

### ISSUE THREAD

**Issue by TimoHocker** [Rem0o/FanControl.Releases#2046]
**Describe the bug**

Whenever I try to shutdown my computer it tells me that an unnamed program with a generic icon is preventing shutdown. The message will stay like this for about one minute until Windows gives up and returns to the login screen. This makes it impossible for me to remotely shutdown my computer.
Through the eventviewer I have found out that the responsible program is FanControl.exe and I could confirm it by killing FanControl right before I shutdown the computer.

There is no unsaved configuration, if the program is requested to exit through the tray icon, it does so normally.
Looking through other issues I have noticed that other people are having this problem from time to time, though nothing I found helped to resolve my issue.

I would like to suggest that FanControl should exit if requested by Windows regardless if there are unsaved changes, save them in a second file or discard them if they are only changes due to an update or added sensors. As it seems this is not the root cause of my system not being able to shut down, still there is no reason for FanControl to ever prevent my computer from shutting down.

**Relevant hardware specs and setup**
System: Windows 11
CPU: AMD Ryzen 5900X
RAM: 32GB 3200MHz DDR4
GPU: GTX 1660
Storage:
- NVME System Storage
- NVME Game Storage
- 4 mixed old SATA HDDs and SSDs in Spanned Volume as Backup Storage

Only Motherboard, CPU, GPU and NvApiWrapper are enabled in Sensor settings.

**Comment by Rem0o** [comment]
The only way FanControl can prevent shutdown is a "unsaved changes" dialog. There is no other way it can do that. So I would be pretty confidant that was what you saw.

**Comment by Rem0o** [comment]
@TimoHocker  if you cancel the shutdown and look at the FanControl window, what do you get?

**Comment by TimoHocker** [comment]
When I cancel the blocked shutdown, FanControl is gone from the tray. If I then clich shutdown again, it successfully shuts down.
Through Eventviewer I could again confirm that FanControl was blocking shutdown even though it appears to have successfully closed.
I found something else: The problem only occurs when FanControl is in the tray. When I open the window and then click shutdown, it works fine.

**Comment by Rem0o** [comment]
Any logs in your log.txt file?

**Comment by TimoHocker** [comment]
The log only contains NvAPI disconnects due to driver updates, last one 6 days ago.

**Comment by TimoHocker** [comment]
That sounds like it could possibly cause that issue. Though there should not be much it has to wait for since I disabled all unnecessary sources.
I disabled NvApiWrapper to see if that would change anything, but it still looks the same. Is there a way to get verbose logging or can you provide a build with detailed logging in the appropriate spots?

**Comment by Rem0o** [comment]
[TimoHockerLogs.zip](https://github.com/Rem0o/FanControl.Releases/files/12744297/TimoHockerLogs.zip)

Put "some" logs at relevant spots.  Shoot those in your folder ( assuming you got the 4.8 version ), and go through a shutdown and we'll see.

**Comment by TimoHocker** [comment]
The log says it closes immediately.

```log
28 Sep 2023 08:33:34: TimoHocker: Session ending.
28 Sep 2023 08:33:34: TimoHocker: State has NOT changed, closing.
28 Sep 2023 08:33:34: TimoHocker: Closing LibreHardwareMonitor...
28 Sep 2023 08:33:34: TimoHocker: close success!
28 Sep 2023 08:40:36: TimoHocker: Session ending.
28 Sep 2023 08:40:36: TimoHocker: State has NOT changed, closing.
28 Sep 2023 08:40:36: TimoHocker: Closing LibreHardwareMonitor...
28 Sep 2023 08:40:36: TimoHocker: close success!
```

In the eventviewer I found that Docker Desktop was getting terminated by windows for not responding. Then I tried terminating Docker myself before shutting down and it actually did without waiting. It seems like there has to be some conflict between Docker and FanControl because killing either process allows the system to shut down.
Since Docker was acting up with dozens of errors when closing and I don't need it I decided to uninstall Docker and WSL. Now the problem is gone.
My conclusion: Docker Desktop is a pile of garbage that doesn't know how to close itself and Windows is stupid enough to blame it on other programs.

**Comment by Rem0o** [comment]
Tried Docker for windows once, it was indeed garbage. 

Glad we got to the bottom of this, good occasion to review.

Closing.

**Comment by theotherp** [comment]
By the way, as I got here for the same reason (misidentified reason for hanging shutdown): This was fixed with Docker Desktop 4.25: https://github.com/docker/for-win/issues/13697#issuecomment-1780808079
